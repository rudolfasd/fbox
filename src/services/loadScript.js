const loadScript = (scriptSrc, scriptId, callback) => {
    const existingScript = document.getElementById(scriptId);
    if (!existingScript) {
        const script = document.createElement('script');
        script.src = scriptSrc;
        script.id = scriptId;
        document.head.appendChild(script);
        script.onload = () => {
            callback();
        };
    };
};

export default loadScript;