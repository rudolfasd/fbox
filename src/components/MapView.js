import React from 'react';

import loadScript from '../services/loadScript';
import ymapCreate from '../services/ymap';


const MAP_PARENT_ELEMENT = 'MapView';
const scriptSrc = process.env.REACT_APP_YMAPS_SCRIPT
    .replace('<apikey>',process.env.REACT_APP_YMAPS_API_KEY);
const scriptId = 'YMaps';

export const scriptOnloadCallback = () => ymapCreate(
    MAP_PARENT_ELEMENT,
    {
        center: [57, 95],
        zoom: 3
    },{
        searchControlProvider: 'yandex#search',
        balloonPanelMaxMapArea: Infinity
    }
)


class MapView extends React.Component {

    componentDidMount(){
        loadScript(scriptSrc, scriptId, scriptOnloadCallback);
    }

    render(){
        return(
            <div className='map-view' id={MAP_PARENT_ELEMENT}></div>
        );
    }
}

export default MapView;

