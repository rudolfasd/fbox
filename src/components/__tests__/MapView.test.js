import React from 'react';

import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import MapView from '../MapView';
import loadScript from '../../services/loadScript';
import ymapCreate from '../../services/ymap';


jest.mock('../../services/loadScript');
jest.mock('../../services/ymap');


describe('REACT COMPONENT <MapView />',() => {

    beforeEach(()=>{
        loadScript.mockClear();
    });

    it('MapView - snapshot',() => {
        const renderedValue = renderer.create(
            <MapView />
        ).toJSON();
        expect(renderedValue).toMatchSnapshot();
    }); 

    it('MapView - componentDidMount should call loadScript', () => {
        shallow(<MapView />);
        expect(loadScript).toHaveBeenCalledTimes(1);
    });

    it('MapView - loadScript should call ymapCreate', () => {
        loadScript.mockImplementation((scriptSrc,scriptId,callback) => {
            return (
                require.requireActual(
                    '../../services/loadScript'
                ).default
            )(scriptSrc,scriptId,callback);
        });
        shallow(<MapView />);
        document.querySelector('script#YMaps').onload();
        expect(ymapCreate).toHaveBeenCalledTimes(1);
    });
});