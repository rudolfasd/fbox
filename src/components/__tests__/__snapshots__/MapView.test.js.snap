// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`REACT COMPONENT <MapView /> MapView - snapshot 1`] = `
<div
  className="map-view"
  id="MapView"
/>
`;
